py3status/modules/process_status.py

```diff
@@ -3,68 +3,85 @@
 Display if a process is running.
 
 Configuration parameters:
-    cache_timeout: how often to run the check (default 10)
-    format_not_running: what to display when process is not running
-        (default '■')
-    format_running: what to display when process running (default '●')
-    full: if True, match against the full command line and not just the
-        process name (default False)
-    process: the process name to check if it is running (default None)
+    cache_timeout: refresh interval for this module (default 5)
+    format: default format for this module (default '{icon}')
+    full: if True, match against full command line (default False)
+    icon_off: show when process not running (default '■')
+    icon_on: show when process running (default '●')
+    process: process name to check for (default None)
+
+Format placeholders:
+    {icon} process icon
+    {process} process name
 
 Color options:
-    color_bad: Process not running or error
-    color_good: Process running
+    color_bad: Not running
+    color_good: Running
 
 @author obb, Moritz Lüdecke
 """
-
-import os
-import subprocess
-
-ERR_NO_PROCESS = 'no process name given'
+string_error = 'process_status: N/A'
 
 
 class Py3status:
     """
     """
     # available configuration parameters
-    cache_timeout = 10
-    format_not_running = u'■'
-    format_running = u'●'
+    cache_timeout = 5
+    format = '{icon}'
     full = False
+    icon_off = u'■'
+    icon_on = u'●'
     process = None
 
-    def _get_text(self):
-        fnull = open(os.devnull, 'w')
-        pgrep = ["pgrep", self.process]
-
-        if self.full:
-            pgrep = ["pgrep", "-f", self.process]
+    class Meta:
+        deprecated = {
+            'rename': [
+                {
+                    'param': 'format_running',
+                    'new': 'icon_on',
+                    'msg': 'obsolete parameter use `icon_on`',
+                },
+                {
+                    'param': 'format_not_running',
+                    'new': 'icon_off',
+                    'msg': 'obsolete parameter use `icon_off`',
+                },
+            ],
+        }
 
-        if subprocess.call(pgrep,
-                           stdout=fnull, stderr=fnull) == 0:
-            text = self.format_running
-            color = self.py3.COLOR_GOOD
+    def post_config_hook(self):
+        self.color_on = self.py3.COLOR_ON or self.py3.COLOR_GOOD
+        self.color_off = self.py3.COLOR_OFF or self.py3.COLOR_BAD
+
+    def _is_running(self):
+        try:
+            pgrep = ["pgrep", self.process]
+            if self.full:
+                pgrep = ["pgrep", "-f", self.process]
+            self.py3.command_output(pgrep)
+        except:
+            return False
         else:
-            text = self.format_not_running
-            color = self.py3.COLOR_BAD
-
-        return (color, text)
+            return True
```

Comment:
here you could just `return True` after `self.py3.command_output(pgrep)` and then we don't need the else - I don't think they read very nicely